en/operations/admin-procedures.html

```diff
@@ -121,6 +121,217 @@ <h2 id="status-pages">Status pages</h2>
+<p>
+  Add index aspect and match/stemming settings to the field, deploy and observe output
+</p>
+<pre>
+field artist type string {
+  indexing: summary | attribute <span style="background-color: yellow;">| index
+    match    : word
+    stemming : none</span>
+}
+
+$ (cd src/main/application && zip -r - .) |   curl --header Content-Type:application/zip --data-binary @- \
+  localhost:19071/application/v2/tenant/default/prepareandactivate
+
+{
+  "log": [
+    {
+      "time": 1628239290150,
+      "level": "WARNING",
+      "message": "Change(s) between active and new application that may require re-index:\nindexing-change:
+                  Consider re-indexing document type 'music' in cluster 'music' because:\n
+                  1) Document type 'music': Field 'artist' changed: add index aspect,
+                  indexing script: '{ input artist | summary artist | attribute artist; }' ->
+                                   '{ input artist | exact | summary artist | attribute artist | index artist; }'\n"
+    }
+  ],
+  "tenant": "default",
+  "url": "http://localhost:19071/application/v2/tenant/default/application/default/environment/prod/region/default/instance/default",
+  "message": "Session 3 for tenant 'default' prepared and activated.",
+  "configChangeActions": {
+    "restart": [],
+    "refeed": [],
+    "reindex": [
+      {
+        "name": "indexing-change",
+        "documentType": "music",
+        "clusterName": "music",
+        "messages": [
+          "Document type 'music': Field 'artist' changed: add index aspect, indexing script: '{ input artist | summary artist | attribute artist; }' -> '{ input artist | exact | summary artist | attribute artist | index artist; }'"
+        ],
+        "services": [
+          {
+            "serviceName": "searchnode",
+            "serviceType": "searchnode",
+            "configId": "music/search/cluster.music/0",
+            "hostName": "vespa-container"
+          }
+        ]
+      }
+    ]
+  }
+}
+</pre>
+<p>
+  Trigger reindexing, deploy again, wait until done
+  (look for the log message / use the <code>reindexing</code> endpoint).
+  Dumping the index structures now show artist both in index and attribute.
+  Verify the query still works:
+</p>
+<pre>
+$ curl http://localhost:19071/application/v2/tenant/default/application/default/environment/prod/region/default/instance/default/reindex
+
+{
+  "generation": 3,
+  "applicationPackageFileReference": "",
+  "modelVersions": ["7.423.37"]
+}
+
+$ curl http://localhost:19071/application/v2/tenant/default/application/default/environment/prod/region/default/instance/default/reindexing
+{
+  "enabled": true,
+  "clusters": {
+    "music": {
+      "pending": {
+        "music": 3
+      },
+      "ready": {
+        "music": {}
+      }
+    }
+  }
+}
+
+$ (cd src/main/application && zip -r - .) |   curl --header Content-Type:application/zip --data-binary @- \
+  localhost:19071/application/v2/tenant/default/prepareandactivate
```

Comment:
Hmm, why not use `vespa-deploy`? I thought that was the most common use? The `/application/v2/.../reindexing` API was intended for manually triggering reindexing, when using that deploy tool. `prepareandactivate` indeed lets the change be automatically detected, and acted upon, but I don't think that's even in the opensource documentation—neither of them?

Reply:
... not saying it shouldn't be, though.

Reply:
Using `prepareandactivate` makes it more similar to Vespa Cloud, easier to swap between the two. See https://docs.vespa.ai/en/cloudconfig/deploy-rest-api-v2.html#prepareandactivate . This doc needs to be updated as it is more than a prepare/activate (as it actually will do reindex), will do this soon

Reply:
ok - so maybe we need a new deploy api command then as https://docs.vespa.ai/en/cloudconfig/deploy-rest-api-v2.html requires a session (unless one can use * or something as session ID to use latest / create one auto)..

We can discuss that separately, I'll create a ticket

Reply:
> activate without prepare does an implicit prepare

I don't think that's true, for the HTTP API—looking at the code, the activate endpoint explicitly creates a `Deployment.prepared(...)`, which is then activated.